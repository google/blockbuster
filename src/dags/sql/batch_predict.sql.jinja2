SELECT
  SUBSTR(cid, 3) AS cid,
  'event' AS t,
  {{ event_label }} AS el,
  {{ event_action }} AS ea,
  {{ event_category }} AS ec,
  NTILE({{ num_segments }}) OVER (ORDER BY {{ event_label }} ASC) AS ev
FROM (
  SELECT
    BB_id AS cid,
    RANK() OVER (PARTITION BY BB_id ORDER BY effectiveDate DESC) as rank,
    ROUND(t1.tables.score, 5) AS {{ event_label }}
  FROM `{{ source_table }}`,
  UNNEST(predicted_predictionLabel) AS t1
  WHERE t1.tables.value='true'
)
WHERE rank = 1;
