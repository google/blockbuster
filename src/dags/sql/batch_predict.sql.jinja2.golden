SELECT
  SUBSTR(cid, 3) AS cid,
  'event' AS t,
  score AS el,
  'action' AS ea,
  'category' AS ec,
  NTILE(50) OVER (ORDER BY score ASC) AS ev
FROM (
  SELECT
    BB_id AS cid,
    RANK() OVER (PARTITION BY BB_id ORDER BY effectiveDate DESC) as rank,
    ROUND(t1.tables.score, 5) AS score
  FROM `project.dataset.table`,
  UNNEST(predicted_predictionLabel) AS t1
  WHERE t1.tables.value='true'
)
WHERE rank = 1;
